src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -121,6 +98,56 @@ case class CsvRelation protected[spark] (
     new InterpretedMutableProjection(casts, startSchema)
   }
 
+  private def readCSV(
+     file: RDD[String],
+     header: Seq[String],
+     schemaFields: Seq[StructField],
+     projection: MutableProjection,
+     row: GenericMutableRow) = {
+
+    val dataLines = if (useHeader) {
+      file.mapPartitionsWithIndex({
+        case (partitionIndex, iter) => if (partitionIndex == 0) iter.drop(1) else iter
```

Comment:
One example, is reading files generated with previous versions of spark-csv. We put the header line in each file (effectively each partition output). With this change users who wrote data with previous versions of spark-csv cannot properly read it back.